balancer-js/src/modules/pools/apr/apr.integration.spec.ts

```diff
@@ -0,0 +1,105 @@
+import dotenv from 'dotenv';
+import { expect } from 'chai';
+import MockDate from 'mockdate';
+import { BalancerSDK } from '@/modules/sdk.module';
+import { JsonRpcProvider } from '@ethersproject/providers';
+
+dotenv.config();
+
+const sdk = new BalancerSDK({
+  network: 1,
+  rpcUrl: 'http://127.0.0.1:8545',
+});
+
+const { pools } = sdk;
+
+const ethStEth =
+  '0x32296969ef14eb0c6d29669c550d4a0449130230000200000000000000000080';
+
+const veBalId =
+  '0x5c6ee304399dbdb9c8ef030ab642b10820db8f56000200000000000000000014';
+
+const usdStable =
+  '0x7b50775383d3d6f0215a8f290f2c9e2eebbeceb20000000000000000000000fe';
+
+const btcEth =
+  '0xa6f548df93de924d73be7d25dc02554c6bd66db500020000000000000000000e';
+
+const ethStEthCopy =
+  '0x851523a36690bf267bbfec389c823072d82921a90002000000000000000001ed';
+
+describe('happy case', () => {
+  // Time when veBal used to recieve procotol revenues
+  const now = new Date('2022-08-19 11:11:11').getTime();
+
+  before(async function () {
+    this.timeout(2e4);
+
+    MockDate.set(now);
+
+    const rpcProvider = sdk.rpcProvider as JsonRpcProvider;
+
+    await rpcProvider.send('hardhat_reset', [
+      {
+        forking: {
+          jsonRpcUrl: process.env.RPC_URL || process.env.ALCHEMY_URL,
+          blockNumber: 15370937, // 2022-08-19 11:11:11
+        },
+      },
+    ]);
+  });
+
+  after(() => {
+    MockDate.reset();
+  });
+
+  describe('duplicated pool with an empty gauge', () => {
+    it('has tokenAprs', async () => {
+      const pool = await pools.find(ethStEthCopy);
+      if (pool) {
+        const { apr } = pool;
+        expect(apr && apr.tokenAprs).to.be.greaterThan(1);
+      }
+    }).timeout(120000);
```

Comment:
when something throws the test stops. timeout is setup to wait for network connections. it might get slow on the first run. however it will be faster once cached